doc/src/read_data.rst

```diff
@@ -654,7 +654,7 @@ The per-atom values have these meanings and units, listed alphabetically:
 * sp = magnitude of magnetic spin of atom (Bohr magnetons)
 * spx,spy,spz = components of magnetic spin of atom (unit vector)
 * template-atom = which atom within a template molecule the atom is
-* template-index = which molecule within the molecule template the atom is part of
+* template-index = which molecule within the molecule template (0-based) the atom is part of
```

Comment:
We should change this to 1-based in the data file, but store the indices as 0-based internally.  Likewise data file values of 0 should mean that the atom is not part of a molecule.  Similar to ellipsoidflag.  That should be documented here.  Will require subtracting 1 from the read-in values in data_atom_post().

Reply:
this is implemented and several changes implemented to avoid false positives in error and segmentation faults.